src/components/head-metadata/index.tsx

```diff
@@ -0,0 +1,47 @@
+import HashiHead from '@hashicorp/react-head'
+import { useCurrentProduct } from 'contexts'
+import { HeadMetadataProps } from './types'
+
+/**
+ * Builds up the the necessary meta tags for the site. Rendered in `_app`, where it receives `pageProps.metadata` as props
+ */
+export default function HeadMetadata(props: HeadMetadataProps) {
+  const { name: productName } = useCurrentProduct() ?? {}
+
+  const titleParts = [__config.dev_dot.meta.title]
+  const description = props.description ?? __config.dev_dot.meta.description
+
+  if (productName) {
+    titleParts.unshift(productName)
+  }
+
+  if (props.title) {
+    titleParts.unshift(props.title)
+  }
+
+  // Ultimately we want to build up a title that looks like {specified page title} | {product} | {root title}
```

Comment:
💭 What do you think about putting this comment in the block comment above `HeadMetadata` or creating a helper function just for building the title?